src/asm2wasm.h

```diff
@@ -1537,6 +1537,7 @@ void Asm2WasmBuilder::optimize() {
   passRunner.add("remove-unused-names");
   passRunner.add("merge-blocks");
   passRunner.add("simplify-locals");
+  passRunner.add("constant-propagation");
```

Comment:
for asm2wasm, technically we shouldn't need this - if the code is built with `-Oz`. The opportunities your pass finds are because they code in the test suite is `-O0` or `-O2`, and that focuses less on code size, and in particular it duplicates blocks, which leads to ssa optimizations not kicking in specifically on return values.

but, since `-O2` is common, I suppose this does make sense. just writing some thoughts here, not  request to change anything